nkms/blockchain/eth/sol/source/contracts/MinersEscrow.sol

```diff
@@ -423,53 +366,76 @@ contract MinersEscrow is Issuer {
     function mint() public onlyTokenOwner {
         uint256 previousPeriod = getCurrentPeriod().sub(uint(1));
         MinerInfo storage info = minerInfo[msg.sender];
-        uint256 numberPeriodsForMinting = info.confirmedPeriods.length;
-        if (numberPeriodsForMinting == 0 || info.confirmedPeriods[0].period > previousPeriod) {
+
+        if (info.confirmedPeriod1 > previousPeriod &&
+            info.confirmedPeriod2 > previousPeriod ||
+            info.confirmedPeriod1 > previousPeriod &&
+            info.confirmedPeriod2 == EMPTY_CONFIRMED_PERIOD ||
+            info.confirmedPeriod2 > previousPeriod &&
+            info.confirmedPeriod1 == EMPTY_CONFIRMED_PERIOD ||
+            info.confirmedPeriod1 == EMPTY_CONFIRMED_PERIOD &&
+            info.confirmedPeriod2 == EMPTY_CONFIRMED_PERIOD) {
             return;
         }
 
-        uint256 currentLockedValue = getLockedTokens(msg.sender);
-        ConfirmedPeriodInfo storage last = info.confirmedPeriods[numberPeriodsForMinting - 1];
-        uint256 allLockedPeriods = last.lockedValue
-            .divCeil(info.releaseRate)
-            .sub(uint(1))
-            .add(numberPeriodsForMinting);
+        uint256 first;
+        uint256 last;
+        if (info.confirmedPeriod1 > info.confirmedPeriod2) {
```

Comment:
This optimization (with `confirmedPeriod1` / `confirmedPeriod2`) needs a comment. I am not 100% sure I understand how it works!